lib/recurly/apple-pay/apple-pay.js

```diff
@@ -398,7 +392,7 @@ class ApplePay extends Emitter {
    */
   error (...params) {
     let err = params[0] instanceof Error ? params[0] : errors(...params);
-    this.emit('error', err);
+    setTimeout(() => this.emit('error', err), 0);
```

Comment:
That's a good callout.
The issue is when the user instantiates ApplePay and then tries to start listening for errors.

```js
const applePay = recurly.ApplePay(configOption);
applePay.listen('error', (error) => {/* do something with the error */ });
```

If we emit the initialization error sync, He will not have time to subscribe (line 2), because the error was already emitted.

Notice this does not always happen, it depends on how the initialization error occurs.
It could be async when trying to resync with RA (in this case the prev code works OK), but the error could be raised because something is missing in the config object, in this case, the error is emitted sync and there is no time to subscription

Reply:
Awesome, that makes sense! I'm wondering if we need this same pattern elsewhere in RJS as well, just haven't caught it haha. Will keep it in mind when working in RJS in the future.

Reply:
for sure, we will need to keep 👀 on that.